packages/components/src/templates/next/components/complex/Map/Map.tsx

```diff
@@ -0,0 +1,59 @@
+import type { MapProps } from "~/interfaces"
+import { MAPS_EMBED_URL_PATTERN } from "~/utils/validation"
+import { ComponentContent } from "../../internal/customCssClass"
+
+const isValidGoogleMapsEmbedUrl = (urlObject: URL) => {
+  return (
+    urlObject.hostname === "www.google.com" &&
+    urlObject.pathname === "/maps/embed"
+  )
+}
+
+const isValidOneMapEmbedUrl = (urlObject: URL) => {
+  return (
+    urlObject.hostname === "www.onemap.gov.sg" &&
+    urlObject.pathname === "/minimap/minimap.html"
+  )
+}
+
+const isValidMapEmbedUrl = (url: string) => {
+  if (!url) {
+    return false
+  }
+
+  try {
+    const urlObject = new URL(url)
+
+    return (
+      (isValidGoogleMapsEmbedUrl(urlObject) ||
+        isValidOneMapEmbedUrl(urlObject)) &&
+      new RegExp(MAPS_EMBED_URL_PATTERN).test(url)
```

Comment:
is there a need for the first 2 checks if we are aldy gonna do a regexp against `MAPS_EMBED_URL_PATTERN`? seems like double effort + duplicating of logic in 2 sources

Reply:
Hmm it is very hard to get the regex correct + secure and hence I wanted to have this separate function to correctly parse the URLs with the correct hostname and pathnames. The regex is still there so that we can have some baseline validation done on Studio's side.

Reply:
i see

hmmm... can we then move the functions and constants to `/packages/components/src/utils/validation.ts` so they are in one same place, so that its less likely we will forget to update them in one place but not the other?

Reply:
Updated in e9db158205e42894f882437e1e2063d883a5384e!